shared/src/main/scala/org/widok/validation/Validation.scala

```diff
@@ -0,0 +1,142 @@
+package org.widok.validation
+
+abstract class FieldValidation[T](params: Map[String, Any] = Map.empty, message: String = "validation failed") {
+
+  /**
+   * Implement this method in subclasses to validate a value.
+   *
+   * @param value The value to be validated
+   * @return Either a boolean indicating if validation is successful or an Optional string with a
+   *         validation error message. If a boolean (false) is returned an error message will be constructed using the
+   *         formatMessage method.
+   */
+  protected def validate(value: T): Either[Boolean, Option[String]]
+
+  def validateValue(value: Any): Option[String] = {
+    val typedValue: T = value.asInstanceOf[T]
+
+    validate(typedValue) match {
+      case Left(result) => if (result) None else Some(formatMessage(message, params, typedValue))
+      case Right(validationMsg) => validationMsg
+    }
+  }
+
+  def formatMessage(msg: String, params: Map[String, Any], value: T) = {
+    val paramsWithValue = params + ("value" -> value)
+    (msg /: paramsWithValue) { (res, entry) => res.replaceAll("#\\{%s\\}".format(entry._1), entry._2.toString.replace( """\""", """\\""")) }
+  }
+}
+
+abstract class TextFieldValidation(params: Map[String, Any] = Map.empty, message: String = "validation failed") extends FieldValidation[String](params, message) {
+  override def validateValue(value: Any): Option[String] = {
+    if (!value.isInstanceOf[String]) throw new IllegalArgumentException("Expected a string value")
+    val typedValue = value.asInstanceOf[String]
+    if (typedValue.isEmpty) None else super.validateValue(typedValue)
+  }
+}
+
+object Validations {
+
+  /**
+   * Validates min values
+   *
+   * @param min The minimum value for the field
+   * @param message Error message. Variables: #{value} the validated input, #{min} = the min value
+   */
+  case class MinValidation(min: Int, message: Option[String] = None)
+    extends TextFieldValidation(Map("min" -> min), message.getOrElse("Value must be at least #{min}")) {
```

Comment:
Is there a way to do this kind of formatting in scala? I've been googling but didn't come up with anything.

I need something like this (example code, won't work ofcourse):

```scala
val m = Map("car" -> "saab", "color" -> "yellow")
"My $car is $color".format(m)
```